### PR TITLE
[[ Bugfix 20951 ]] Fixed bug causing garbage at end of repeat to be ignored

### PR DESCRIPTION
**Pull request by abrisan**

### CODE REVIEW

File: tests/lcs/parser/repeat.parsertest [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+%% You should have received a copy of the GNU General Public License
+%% along with LiveCode.  If not see <http://www.gnu.org/licenses/>.
+
+%% REPEAT WITH no step
+%TEST RepeatWithNoStep
+on mouseUp
+  repeat with i = 1 to 10
+  end repeat
+end mouseUp
+%EXPECT_PASS
```
Comment (livecodeali) [inline tests/lcs/parser/repeat.parsertest]: this needs to be `%EXPECT PASS` (no underscore)

File: tests/lcs/parser/repeat.parsertest [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+%SUCCESS
+%ENDTEST
+
+%% Repeat With step
+%TEST RepeatWithCorrectStep
+on mouseUp
+  repeat with i = 1 to 10 step 1
+  end repeat
+end mouseUp
+%EXPECT_PASS
```
Comment (livecodeali) [inline tests/lcs/parser/repeat.parsertest]: same here

File: tests/lcs/parser/repeat.parsertest [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+%ENDTEST
+
+%% Repeat with step should fail
+%RepeatWithIncorrectStep
+on mouseUp
+  repeat with i = 1 to 10 %{ALEX} 10
+  end repeat
+end mouseUp
+%EXPECT PE_REPEAT_BADCOND AT ALEX
+%SUCCESS
```
Comment (livecodeali) [inline tests/lcs/parser/repeat.parsertest]: This needs to be 
```
%EXPECT PASS
%ERROR PE_REPEAT_BADCOND AT AFTER_ENDVALUE
```

(and rename your label from ALEX to AFTER_ENDVALUE, just so it is more instructive of the intended position). You can still include alex in your garbage if you want, just do it as
```
repeat with i = 1 to 10 %{AFTER_ENDVALUE} alex 10
```
:-)

**Review by livecodeali — CHANGES_REQUESTED** [review]

**Comment by abrisan** [comment]
@livecodeali Should be fixed now

File: tests/lcs/parser/repeat.parsertest [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+%ENDTEST
+
+%% Repeat with step should fail
+%TEST RepeatWithIncorrectStep
+on mouseUp
+  repeat with i = 1 to 10 %{AFTER_ENDVALUE} by 10
+  end repeat
+end mouseUp
+%EXPECT PE_REPEAT_BADCOND AT AFTER_ENDVALUE
+%SUCCESS
```
Comment (livecodeali) [inline tests/lcs/parser/repeat.parsertest]: This bit is still not quite right:
```
%EXPECT PASS
%ERROR PE_REPEAT_BADCOND AT AFTER_ENDVALUE
```

**Comment by abrisan** [comment]
@livecodeali now?

**Comment by livecodeali** [comment]
@livecodealex there shouldn't be a `%SUCCESS` on that one

**Comment by livecodeali** [comment]
Sorry, I didn't make that at all clear originally

**Review by livecodeali — APPROVED** [review]

**Comment by abrisan** [comment]
As @runrevmark nicely pointed out, this is probably failing due to the fact that the IDE tests cannot run. I wonder what we should do in this situation...

**Comment by livecodeali** [comment]
Ah! Well, it probably makes sense to submit a PR fixing the bits in the IDE (this can go into any branch - probably 8.1) and then once it is merged up we can trigger Travis to run again

**Comment by abrisan** [comment]
@livecodeali on it

**Comment by abrisan** [comment]
Testing depends on https://github.com/livecode/livecode-ide/pull/1910

**Comment by livecodeali** [comment]
@livecode-vulcan review ok a9ad5d2

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok a9ad5d2f01c001b2b41be8102d86e2511e35ea7f

**Comment by livecode-vulcan** [comment]
😞 test failure a9ad5d2f01c001b2b41be8102d86e2511e35ea7f

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1422)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1438)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1412)
* try-community-universal-ios-iphoneos11.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.3/builds/15)
* try-community-universal-ios-iphonesimulator11.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.3/builds/15)
* try-community-universal-mac-macosx10.6: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1416)
* try-community-universal-mac-macosx10.9: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1442)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1422)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1436)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1421)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1431)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1472)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1412)

**Comment by livecodepanos** [comment]
@livecode-vulcan retry a9ad5d2

**Comment by livecode-vulcan** [comment]
😎 test success a9ad5d2f01c001b2b41be8102d86e2511e35ea7f

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/303)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/295)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/295)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/295)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/295)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1817)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/45)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/45)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1848)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1838)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1837)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1871)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1812)

**Pull request merged**